codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/ApiGatewayExportsNullabilityExceptionIntegration.java

```diff
@@ -110,150 +138,269 @@ public Model preprocessModel(Model model, GoSettings settings) {
+                model);
+        if (!awsSdkGoV2ChangedNullabilityEvents.isEmpty()) {
+            StringBuilder sb = new StringBuilder().append("AwsSdkGoV2ChangedNullability Validation events found:\n");
+            for (ValidationEvent e : awsSdkGoV2ChangedNullabilityEvents) {
+                sb.append(" " + e.toString() + "\n");
+            }
+            throw new CodegenException(sb.toString());
+        }
+        validateNullabilityExceptions(nullabilityExceptions, model);
+        return model;
     }
 
-    private Model handleApiGateWayExportsNullabilityExceptions(Model model, ShapeId service) {
-        LOGGER.info("Handling APIGW exports nullability exceptions for service: " + service.toString());
-
-        // Read nullability exceptions
+    static Set<ShapeId> getNullabilityExceptions(ToShapeId service) {
+        String nullabilityExceptionsString = IoUtils.readUtf8Resource(
+                ApiGatewayExportsNullabilityExceptionIntegration.class,
+                NULLABILITY_EXCEPTIONS_FILE);
         Set<ShapeId> nullabilityExceptions = Node
-                .parse(IoUtils.readUtf8Resource(getClass(), NULLABILITY_EXCEPTIONS_FILE))
+                .parseJsonWithComments(nullabilityExceptionsString)
                 .expectObjectNode()
-                .expectArrayMember(service.toString())
+                .expectArrayMember(service.toShapeId().toString())
                 .getElementsAs((StringNode s) -> s.expectShapeId())
                 .stream()
                 .collect(Collectors.toSet());
+        return nullabilityExceptions;
+    }
 
-        // Knowledge index
-        Set<Shape> shapesToReplace = new HashSet<>();
-        Set<NumberShape> numberShapes = model.toSet(NumberShape.class);
-        Set<MemberShape> memberShapes = model.getMemberShapes();
+    protected Model getPreviousModel(ShapeId service, Model model) {
+        LOGGER.warning("Getting Previous Model for: " + service.toString());
+        String DIFF_WORKTREE_BRANCH = "__nullability-worktree-" + service;
+        Path root = getRootPath();
 
-        patchDefaultsForRootLevelSnapshottedShapes(
-                model, nullabilityExceptions, shapesToReplace, numberShapes, memberShapes);
-        patchDefaultsForMemberLevelSnapshottedShapes(
-                model, nullabilityExceptions, shapesToReplace, numberShapes, memberShapes);
-        identityNonSnapshottedMemberLevelShapes(
-                nullabilityExceptions, memberShapes);
+        String sdkId = model.getServiceShapesWithTrait(ServiceTrait.class).iterator().next()
+                .expectTrait(ServiceTrait.class).getSdkId()
+                .replace(" ", "-")
+                .toLowerCase();
 
-        // Replace nullability exception shapes
-        return ModelTransformer.create().replaceShapes(model, shapesToReplace);
+        String sha = getPreviousModelSha(root, sdkId, service);
+        LOGGER.warning("Previous model SHA for " + service.toString() + ": " + sha);
```

Comment:
If these log statements going to be the generator for a while I would recommend downgrading these to DEBUG so they don't pollute logs by default. It makes warnings less useful when debug data like these are mixed in.

Reply:
Changed to `LOGGER.info()`.